tools/assets-migration-to-screenly-pro.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import traceback
+import sh
+
+import click
+import requests
+import time
+
+from requests.auth import HTTPBasicAuth
+
+HOME = os.getenv('HOME', '/home/pi')
+
+BASE_API_SCREENLY_URL = 'https://api.screenlyapp.com'
+ASSETS_SCREENLY_OSE_API = 'http://127.0.0.1:8080/api/v1.1/assets'
+
+PORT_NGROK = 4040
+PORT = 8000
+
+ngrok_process = None
+
+token = None
+ngrok_public_url = None
+
+
+################################
+# Suprocesses
+################################
+
+def start_http_ngrok_process(try_connection=100):
+    global ngrok_process
+    click.echo(click.style("Ngrok starting ...", fg='yellow'))
+    ngrok_process = sh.ngrok('http', str(PORT), _bg=True, _in=os.devnull, _out=os.devnull, _err=sys.stderr)
+    try_count = 0
+    while True:
+        if try_count >= try_connection:
+            raise Exception('Failed start ngrok')
+        try:
+            requests.get('http://127.0.0.1:%i' % PORT_NGROK, timeout=10)
+            break
+        except requests.exceptions.ConnectionError:
+            try_count += 1
+            time.sleep(0.1)
+            continue
+    click.echo(click.style("Ngrok successfull started", fg='green'))
+
+
+def get_ngrock_public_url(try_connection=100):
+    try_count = 0
+    while True:
+        if try_count >= try_connection:
+            raise Exception('Could not take a public url ngrok')
+        response = requests.get('http://127.0.0.1:%i/api/tunnels' % PORT_NGROK, timeout=10).json()
+        if response['tunnels']:
+            break
+        else:
+            try_count += 1
+            time.sleep(0.1)
```

Comment:
Probably instead of doing lots of requests linearly - may be it can be less but non-linearly?